docs/tour/adding_capabilities.mdx

```diff
@@ -0,0 +1,232 @@
+
+## Adding more functionality
+
+Let's extend this application to do more than just say "Hello World". We can use the `HttpRequest` struct and check the request for a name provided in a query string, and then return a greeting with that name.
+
+```rust
+async fn handle_request(&self, ctx: &Context, req: &HttpRequest) -> RpcResult<HttpResponse> {
+    let name = match req.query_string.split("=").collect::<Vec<&str>>()[..] {
+        ["name", name] => name.to_string(),
+        _ => "World".to_string(),
+    };
+
+    Ok(HttpResponse::ok(format!("Hello, {}!", name,)))
+}
+```
+
+Using a simple pattern match, we can check to see if we can split the query string on a name parameter, and if so, use that name in the response. If not, we'll just use "World" as the name. This won't support adding multiple query parameters but it's a perfectly good start for our goals.
+
+## Deploying your local actor
+
+Now that we're making modifications to the actor, you'll want to run your local copy instead of the wasmCloud example that's already deployed. To do this, we can edit the manifest to instead point at a local file. You can easily find this path by running `wash build -o json` and looking for **actor_path** in the output. Simply take that and prepend `file://` to the absolute path to avoid any reliance on the current working directory.
+
+```yaml
+metadata:
+  name: hello
+  annotations:
+    version: v0.0.2 # Make sure to upgrade the version
+    description: "wasmCloud Hello World Example"
+spec:
+  components:
+    - name: hello
+      type: actor
+      properties:
+        # Change the image to point at your local file
+        image: file:///Users/wasmcloud/hello/build/docs_s.wasm
+      traits:
+        - type: spreadscaler
+          properties:
+            replicas: 1
+        - type: linkdef
+          properties:
+            target: httpserver
+            values:
+              address: 0.0.0.0:8080
+```
+
+Now you can simply undeploy the old version of the application and deploy the new version, and then you can try out your new feature!
+
+:::info
+Whenever you make a change to your actor that you want to deploy, be sure to run `wash build` to recompile and generate a new .wasm file.
+:::
+
+```bash
+> wash app undeploy hello
+> wash app deploy wadm.yaml
+> curl localhost:8080
+Hello, World!
+> curl 'localhost:8080?name=Bob'
+Hello, Bob!
+```
+
+## Adding persistent storage
+
+To further enhance our application, let's add persistent storage to keep a record of each person that this application greeted. We'll use the key-value store capability for this, and just like HTTP server, you won't need to pick a library or a vendor for your Key-Value Store capability. You'll just need to add the capability to your actor, and then you can pick a provider at runtime.
```

Comment:
"add the capability" -> "add a reference to the capability contract"

this makes it a bit more clear where the division of responsibilities lies